scene/3d/collision_shape_3d.cpp

```diff
@@ -68,46 +68,69 @@ void CollisionShape3D::make_convex_from_siblings() {
 	set_shape(shape_new);
 }
 
+CollisionObject3D *CollisionShape3D::_get_ancestor_collision_object() const {
+	Node *parent = get_parent();
+	while (parent) {
+		CollisionObject3D *co = Object::cast_to<CollisionObject3D>(parent);
+		if (co) {
+			return co;
+		}
+		parent = parent->get_parent();
```

Comment:
If it hits a non-Node3D (such as a viewport), it should abort.

furthermore, it's unclear to me how nodes marked as toplevel should be treated if one exists between the shape and the collisionobject.

Reply:
Thanks, changed, it now aborts as soon as it encounters a non-Node3D node.